testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -787,49 +911,137 @@ impl BitcoinRegtestController {
-            "Miner node: submitting leader_block_commit op for {} - {}, waiting for its inclusion in the next Bitcoin block",
-            &payload.block_header_hash,
-            public_key.to_hex()
+            "Miner node: submitting leader_block_commit (txid: {}, rbf: {}, total spent: {}, size: {}, fee_rate: {})",
+            ongoing_block_commit.txid.to_hex(),
+            ongoing_block_commit.fees.is_rbf_enabled,
+            ongoing_block_commit.fees.total_spent(),
+            ongoing_block_commit.fees.final_size,
+            fee_rate,
         );
 
+        self.ongoing_block_commit = Some(ongoing_block_commit);
+
+        increment_btc_ops_sent_counter();
+
         Some(tx)
     }
 
+    fn build_leader_block_commit_tx(
+        &mut self,
+        payload: LeaderBlockCommitOp,
+        signer: &mut BurnchainOpSigner,
+        _attempt: u64,
+    ) -> Option<Transaction> {
+        // Are we currently tracking an operation?
+        if self.ongoing_block_commit.is_none() {
+            // Good to go, let's build the transaction and send it.
+            let res = self.send_block_commit_operation(payload, signer, &vec![], &vec![], None);
+            return res;
+        }
+
+        let ongoing_op = self.ongoing_block_commit.take().unwrap();
+
+        let _ = self.sortdb_mut();
+        let mined_op = self
+            .burnchain_db
+            .as_ref()
+            .expect("BurnchainDB not opened")
+            .get_burnchain_op(&ongoing_op.txid);
+
+        if mined_op.is_some() {
+            // Good to go, the transaction in progress was mined
+            debug!("Was able to retrieve ongoing TXID - {}", ongoing_op.txid);
+            let res = self.send_block_commit_operation(payload, signer, &vec![], &vec![], None);
+            return res;
+        } else {
+            warn!("Was unable to retrieve ongoing TXID - {}", ongoing_op.txid);
+        }
+
+        // An ongoing operation is in the mempool and we received a new block. The desired behaviour is the following:
+        // 1) If the ongoing and the incoming operation are **strictly** identical, we will be idempotent and discard the incoming.
+        // 2) If the 2 operations are different, we will try to avoid wasting UTXOs, and attempt to RBF the outgoing transaction:
+        //  i) If UTXOs are insufficient,
+        //    a) If no other UTXOs, we'll have to wait on the ongoing operation to be mined before resuming operation.
+        //    b) If we have some other UTXOs, drop the ongoing operation, and track the new one.
+        //  ii) If UTXOs initially used are sufficient for paying for a fee bump, then RBF
+
+        // Let's start by early returning 1)
+        if payload == ongoing_op.payload {
+            debug!("Abort attempt to re-submit identical LeaderBlockCommit");
+            return None;
+        }
+
+        // Let's proceed and early return 2) i)
+        let res = if ongoing_op.fees.estimated_amount_required() > ongoing_op.sum_utxos() {
```

Comment:
This case occurs if the `ongoing_op`'s fees are already too large to increase and replace the transaction via RBF?

Reply:
Naming is probably not optimal - `estimated_amount_required` is returning the estimated amount required for performing a RBF (so ongoing op + fee_rate increase + bandwidth fee).